com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/SessionController.java

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+ * Copyright (c) 2019 EclipseSource and others.
+ *
+ *   This program and the accompanying materials are made available under the
+ *   terms of the Eclipse Public License v. 2.0 which is available at
+ *   http://www.eclipse.org/legal/epl-2.0.
+ *
+ *   This Source Code may also be made available under the following Secondary
+ *   Licenses when the conditions for such availability set forth in the Eclipse
+ *   Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ *   with the GNU Classpath Exception which is available at
+ *   https://www.gnu.org/software/classpath/license.html.
+ *
+ *   SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ *******************************************************************************/
+package com.eclipsesource.modelserver.emf.common;
+
+import static java.util.stream.Collectors.toSet;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.eclipse.emf.ecore.EObject;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
+
+import com.eclipsesource.modelserver.jsonschema.Json;
+import com.google.common.collect.Maps;
+import com.google.inject.Inject;
+
+import io.javalin.websocket.WsContext;
+import io.javalin.websocket.WsHandler;
+
+public class SessionController extends WsHandler {
+
+	private Map<String, Set<WsContext>> modelUrisToClients = Maps.newConcurrentMap();
+
+	@Inject
+	private ModelRepository modelRepository;
+
+	public void subscribe(WsContext ctx, String modeluri) {
+		modelUrisToClients.computeIfAbsent(modeluri, clients -> ConcurrentHashMap.newKeySet()).add(ctx);
+		ctx.send(JsonResponse.success(Json.object(Json.prop("sessionId", Json.text(ctx.getSessionId())))));
+	}
+
+	public void unsubscribe(WsContext ctx) {
+		modelUrisToClients.forEach((modeluri, clients) -> {
+			clients.remove(ctx);
+			if (clients.isEmpty())
```

Comment:
This looks like it should throw an ConcurrentModificationException (it doesn't :) ), but makes me feel uncomfortable. 
Use curly braces for if statement

Reply:
I used the `Iterator` interface now to remove the entry if no clients are left.